### PR TITLE
Add support for PNPM

### LINKED ISSUE #84: pnpm support?
Wondering if there is any chance to get pnpm support?
https://pnpm.io/

### PR DESCRIPTION
**Pull request by ChristopherHaws**
Hey there! This PR adds support for the [PNPM package manager](https://pnpm.io). In order to not add a bunch of if/switch statements, I created a `PackageManager` class which contains the options for each of the package managers. Doing this also removed package manager specific code from many places, so I hope this solution is ok with you! :)

Thanks!

Closes #84

image

### CODE REVIEW

File: src/Resources/pnpm-dark.png
Comment (ChristopherHaws) [inline src/Resources/pnpm-dark.png:1]: I added both the light and dark icons, but I am not sure how to tell the task runner which version to use.

Comment (madskristensen) [reply src/Resources/pnpm-dark.png:1]: I think the task runner explorer only supports a single image.

**Comment by madskristensen** [comment]
Thanks

**Comment by dotnetshadow** [comment]
@ChristopherHaws @madskristensen 
For some reason I can't get this to work. I have **pnpm-lock.yaml**  file in the same directory as package.json. 
Am I missing something?

Is this merged into version 1.5?

**Pull request merged**